shotgun_api3/shotgun.py

```diff
@@ -409,7 +409,7 @@ def __init__(self,
         else:
             self.__ca_certs = os.environ.get('SHOTGUN_API_CACERTS')
 
-        self.base_url = (base_url or "").lower()
+        self.base_url = (base_url or "")
```

Comment:
I am worried that this may introduce regressions. Any client code which compares `sg.base_url` with something will suddenly compare against a potential new value. What's the benefit of removing `lower()`? Urls are case insensitive according to https://tools.ietf.org/html/rfc3986 (page 16):

```
Scheme names consist of a sequence of characters beginning with a
   letter and followed by any combination of letters, digits, plus
   ("+"), period ("."), or hyphen ("-").  Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters.  An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names (e.g., allow "HTTP" as well as "http") for the sake of
   robustness but should only produce lowercase scheme names for
   consistency.
```